test/e2e/metadata_api_test.go

```diff
@@ -95,3 +100,23 @@ func TestMetadataAPI_Fanout(t *testing.T) {
 	testutil.Ok(t, err)
 	testutil.Assert(t, true, len(thanosMeta) == 1 && len(thanosMeta["prometheus_build_info"]) > 0)
 }
+
+func metadataEqual(t *testing.T, meta1, meta2 map[string][]metadatapb.Meta) {
+	// The two responses should have equal # of entries.
+	testutil.Equals(t, len(meta1), len(meta2))
+
+	for metric := range meta1 {
+		// Get metadata for the metric.
+		meta1MetricMeta := meta1[metric]
+		meta2MetricMeta, ok := meta2[metric]
+		testutil.Assert(t, true, ok)
```

Comment:
Just confirming. Don't we need to check `ok` for `meta1MetricMeta` as well?

Reply:
We are iterating meta1 now

Reply:
We are iterating meta1 now